substrate/client/transaction-pool/src/single_state_txpool/single_state_txpool.rs

```diff
@@ -676,8 +688,15 @@ where
 								tx_hash,
 								hash,
 							);
+							Some((
+								// These transactions are coming from retracted blocks, we should
+								// simply consider them external.
+								TimedTransactionSource::new_external(false),
```

Comment:
From what I understand it could be `InBlock`. But I don't understand all the consequences of changing this value, as it goes to runtime side. I'd not touch it in this PR :)